src/components/encoding-pane/property-editor-schema.ts

```diff
@@ -0,0 +1,96 @@
+import * as vlSchema from 'vega-lite/build/vega-lite-schema.json';
+
+export interface PropertyEditorFormatSchema {
+  type: string;
+  title: string;
+  properties: any;
+}
+
+export interface PropertyEditorSchema {
+  uiSchema: any;
+  schema: PropertyEditorFormatSchema;
+}
+
+export function getPropertyEditorSchema(prop: string, nestedProp: string): PropertyEditorSchema {
+  if (prop === 'scale') {
+    if (nestedProp === 'type') {
+      return generateSchema(prop, nestedProp, SCALE_TYPE_UISCHEMA,
```

Comment:
1) Why declaring UISCHEMA below, not up here.

2) UISCHEMA shouldn't contain the key they you automatically generate anymore.